hoomd/md/Thermostat.h

```diff
@@ -0,0 +1,193 @@
+//
+// Created by girard01 on 11/9/22.
+//
+
+#ifndef HOOMD_THERMOSTAT_H
+#define HOOMD_THERMOSTAT_H
+
+#include <array>
+#include <hoomd/HOOMDMath.h>
+#include <hoomd/ParticleGroup.h>
+#include <hoomd/md/ComputeThermo.h>
+#include <hoomd/Variant.h>
+#include "hoomd/RNGIdentifiers.h"
+#include "hoomd/RandomNumbers.h"
+
+namespace hoomd::md {
+
+    class Thermostat {
+    public:
+        Thermostat(std::shared_ptr<ParticleGroup> group,
+                   std::shared_ptr<ComputeThermo> thermo,
```

Comment:
The Python API doesn't expose `ParticleGroup` or the internal thermostat `ComputeThermo` instances. With these arguments in the constructor (good C++ design), this limits us to the string based `thermostat='mttk'` initialization mechanism.

The alternative is to take these arguments out of the constructor so that users can directly instantiate this class in Python (enabling composition): `thermostat.MTTK(kT=1.0, tau=5)`. Then add a `setGroupAndThermo` method that gets called when associated with an integration method. `setGroupAndThermo` should error when called a 2nd time to prevent double usage of a thermostat.